### PR TITLE
feat:mpp-508 add flag for deferred capture in order

### PR DESCRIPTION
**Pull request by joyet-simon**
## Reason for change

[Linear task](https://linear.app/almapay/issue/MPP-xxx/)

### How to test

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: cartridges/int_alma/cartridge/controllers/Alma.js [hunk 1]
```diff
@@ -272,11 +266,11 @@ server.post('CreatePaymentUrl', server.middleware.https, function (req, res, nex
...
     var paymentData = almaPaymentHelper.buildPaymentData(
-            req.querystring.installments,
-            req.querystring.deferred_days,
-            getLocale(req),
-            almaConfigHelper.isDeferredCaptureEnable()
-        );
+        req.querystring.installments,
+        req.querystring.deferred_days,
+        getLocale(req),
+        almaConfigHelper.isDeferredCaptureEnable()
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/controllers/Alma.js:272]: On a défini plus haut que c'est une commande deferred capture, on peut simplifier le buildPaymentData du coup

File: cartridges/int_alma/cartridge/controllers/Alma.js [hunk 2]
```diff
@@ -336,13 +332,20 @@ server.get(
             almaConfigHelper.isDeferredCaptureEnable()
         );
 
+        var isDeferredCapture = almaPaymentHelper.isAvailableForManualCapture(
+            configHelper.isDeferredCaptureEnable(),
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/controllers/Alma.js:336]: On a défini plus haut que c'est une commande deferred capture, on peut simplifier le buildPaymentData du coup

File: cartridges/int_alma/cartridge/controllers/Alma.js (hunk 1 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/controllers/Alma.js]: DONE

File: cartridges/int_alma/cartridge/controllers/Alma.js (hunk 2 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/controllers/Alma.js]: DONE

File: cartridges/int_alma/cartridge/scripts/helpers/almaPaymentHelper.js [hunk 3]
```diff
@@ -367,7 +367,7 @@ function buildPaymentData(installmentsCount, deferredDays, locale, isManualCaptu
         customer: formatCustomerData(currentBasket.getCustomer().profile, currentBasket.getCustomerEmail(), formatAddress(currentBasket.getDefaultShipment().shippingAddress))
     };
 
-    if (isAvailableForManualCapture(isManualCaptureEnabled, installmentsCount, deferredDays)) {
+    if (isManualCaptureEnabled) {
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/scripts/helpers/almaPaymentHelper.js:370]: isManualCapture

**Review by Francois-Gomis — APPROVED** [review]

**Pull request merged**